samples/extensions/raytracing_basic/raytracing_basic.cpp

```diff
@@ -552,38 +586,46 @@ void RaytracingBasic::create_ray_tracing_pipeline()
 	VK_CHECK(vkCreatePipelineLayout(get_device().get_handle(), &pipeline_layout_create_info, nullptr, &pipeline_layout));
 
 	std::array<VkPipelineShaderStageCreateInfo, 3> shader_stages;
-	shader_stages[INDEX_RAYGEN]      = load_shader("khr_ray_tracing_basic/raygen.rgen", VK_SHADER_STAGE_RAYGEN_BIT_KHR);
-	shader_stages[INDEX_MISS]        = load_shader("khr_ray_tracing_basic/miss.rmiss", VK_SHADER_STAGE_MISS_BIT_KHR);
-	shader_stages[INDEX_CLOSEST_HIT] = load_shader("khr_ray_tracing_basic/closesthit.rchit", VK_SHADER_STAGE_CLOSEST_HIT_BIT_KHR);
+	shader_stages[0] = load_shader("khr_ray_tracing_basic/raygen.rgen", VK_SHADER_STAGE_RAYGEN_BIT_KHR);
+	shader_stages[1] = load_shader("khr_ray_tracing_basic/miss.rmiss", VK_SHADER_STAGE_MISS_BIT_KHR);
+	shader_stages[2] = load_shader("khr_ray_tracing_basic/closesthit.rchit", VK_SHADER_STAGE_CLOSEST_HIT_BIT_KHR);
 
 	/*
 		Setup ray tracing shader groups
 	*/
-	std::array<VkRayTracingShaderGroupCreateInfoKHR, 3> groups{};
-	for (auto &group : groups)
-	{
-		// Init all groups with some default values
-		group.sType              = VK_STRUCTURE_TYPE_RAY_TRACING_SHADER_GROUP_CREATE_INFO_KHR;
-		group.generalShader      = VK_SHADER_UNUSED_KHR;
-		group.closestHitShader   = VK_SHADER_UNUSED_KHR;
-		group.anyHitShader       = VK_SHADER_UNUSED_KHR;
-		group.intersectionShader = VK_SHADER_UNUSED_KHR;
-	}
-
-	// Links shaders and types to ray tracing shader groups
-	groups[INDEX_RAYGEN].type                  = VK_RAY_TRACING_SHADER_GROUP_TYPE_GENERAL_KHR;
-	groups[INDEX_RAYGEN].generalShader         = INDEX_RAYGEN;
-	groups[INDEX_MISS].type                    = VK_RAY_TRACING_SHADER_GROUP_TYPE_GENERAL_KHR;
-	groups[INDEX_MISS].generalShader           = INDEX_MISS;
-	groups[INDEX_CLOSEST_HIT].type             = VK_RAY_TRACING_SHADER_GROUP_TYPE_TRIANGLES_HIT_GROUP_KHR;
-	groups[INDEX_CLOSEST_HIT].closestHitShader = INDEX_CLOSEST_HIT;
+	VkRayTracingShaderGroupCreateInfoKHR raygen_group_ci{};
+	raygen_group_ci.sType              = VK_STRUCTURE_TYPE_RAY_TRACING_SHADER_GROUP_CREATE_INFO_KHR;
+	raygen_group_ci.type               = VK_RAY_TRACING_SHADER_GROUP_TYPE_GENERAL_KHR;
+	raygen_group_ci.generalShader      = 0;
```

Comment:
can we use the new defines (e.g. INDEX_RAYGEN_GROUP) here too?

Reply:
In theory yes, but it could cause confusion, as the index of the shader (in the pipeline) is not the same as the group index. They only match in this sample because there are as many groups as shaders and they're in the same order. So I'd rather not use the _GROUP constants here as this may confuse people that try to use more complex or different ray tracing shader group setups.

Reply:
You could make different defines for stages and groups and postpend _GROUP and _STAGE. I'm a fan of not using literals as identifiers.

Reply:
That's true. That would make the sample more consistent.